### PR TITLE
Added grpc_cli dockerfile

### PR DESCRIPTION
**Pull request by mdkess**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,15 +54,49 @@ $ docker run ... namely/protoc-all -f protorepo/catalog/catalog.proto -l go
...
+alias grpc_cli='docker run -v `pwd`:/defs --rm -it namely/grpc-cli'
+```
+
+Note the use of single quotes in the alias, which avoids expanding the `pwd` parameter when the alias
+is created.
+
+Now you can call it with
+
+```sh
+grpc_cli call docker.for.mac.localhost:50051 LinkShortener.ResolveShortLink "short_link:'asdf'" --protofiles=link_shortener.proto
```
Comment (mhamrah) [inline README.md]: wortwhile adding a multi-file example to avoid users googling.

File: cli/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+# Grab standard proto libraries
+WORKDIR /tmp
+RUN git clone https://github.com/google/protobuf.git
+RUN git clone https://github.com/googleapis/googleapis
+RUN git clone https://github.com/grpc-ecosystem/grpc-gateway
+
+
+# Build release image
+FROM debian
```
Comment (mhamrah) [inline cli/Dockerfile]: protoc-all uses alpine. Any reason for the switch?

File: cli/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+WORKDIR /tmp/grpc
+RUN git submodule update --init
+
+# Build grpc_cli
+# For some reason, we have to do this manually.
+WORKDIR /tmp/grpc/third_party/protobuf
+RUN ./autogen.sh
+
+WORKDIR /tmp/grpc
+RUN make grpc_cli
```
Comment (mhamrah) [inline cli/Dockerfile]: Why not run this in https://github.com/namely/docker-protoc/blob/master/all/Dockerfile and just create a new dockerfile with protoc-all as base and a new entrypoint? You'd remove this file and duplication completely.

File: cli/Dockerfile (hunk 2 above)
Comment (mdkess) [reply cli/Dockerfile]: Yeah - I couldn't find libgflags-dev for Alpine, which is required by grpc. That's also the answer to the other question.

File: README.md (hunk 1 above)
Comment (mdkess) [reply README.md]: Good call! Done!

File: cli/Dockerfile (hunk 2 above)
Comment (mhamrah) [reply cli/Dockerfile]: You can grab it from a mirror, see https://github.com/tombh/texttop/blob/master/Dockerfile.

Comment (mhamrah) [reply cli/Dockerfile]: may be better: https://gist.github.com/dukelion/2a8813f8adb4ac08b500175f4daa88fd

**Comment by mdkess** [comment]
As discussed - I'll submit this as is, but I'll try to go back and consolidate stuff later.

**Pull request merged**